examples/slack-connector/src/main/java/com/salesforce/connector/slack/service/SlackService.java

```diff
@@ -0,0 +1,221 @@
+import com.slack.api.model.block.element.StaticSelectElement;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+
+@Service
+public class SlackService {
+
+  private static final Logger logger = LoggerFactory.getLogger(SlackService.class);
+
+  @Autowired
+  private ChatbotService chatbotService;
+
+  private String replyInThreadChannels;
+
+  @PostConstruct
+  public void initialize(){
+
+    if(System.getenv(Constants.REPLY_IN_THREAD_CHANNELS)!=null){
+      this.replyInThreadChannels = System.getenv(Constants.REPLY_IN_THREAD_CHANNELS);
+    } else {
+      this.replyInThreadChannels = "";
+    }
+  }
+
+  public void addThreadToReplyInThread(String channelId){
+    if(this.replyInThreadChannels.contains(channelId)){
+      this.replyInThreadChannels = this.replyInThreadChannels.replace(channelId,
+          Constants.EMPTY_STRING);
+    }else {
+      this.replyInThreadChannels += "," + channelId;
+    }
+  }
+
+  public boolean isFlag(String key){
+    return (System.getenv(key)!=null && System.getenv(key).equalsIgnoreCase("true"));
+  }
+
+  public boolean isReplyInThreadChannel(String channel){
+    return (this.replyInThreadChannels!=null && this.replyInThreadChannels.contains(channel));
+  }
+
+  public String cleanTextResponse(String responseText, String user){
+    return responseText.replace(Constants.BOT_RESPONSE_CLEANUP_MATCH,"<@"+ user + ">");
+  }
+
+  public String cleanTextRequest(String text, boolean isAppMention){
+    String response = text;
+    if (isAppMention) {
+      response = response.substring(14);
+    }
+    return response.trim();
+  }
+
+  private void processTextResponse(AnyResponseMessage message, Context ctx, String user,
+      String channel, String threadTs ) throws SlackApiException, IOException {
+    logger.info("MESSAGE: {}", ((TextResponseMessage) message).getText());
+    String response = cleanTextResponse(((TextResponseMessage) message).getText(),
+        user);
+    ctx.client().chatPostMessage(r -> r
+        .channel(channel)
+        .text(response)
+        .threadTs(threadTs)
+        .parse("full")
+        .unfurlLinks(true)
+        .unfurlMedia(true));
+
+  }
+
+  private void processChoiceResponse(AnyResponseMessage message, Context ctx, String channel,
+      String threadTs) throws SlackApiException, IOException {
+    logger.info("MESSAGE: **choice message**");
+    java.util.List<com.slack.api.model.block.LayoutBlock> messageBlocks;
+    if (isFlag(Constants.USE_BUTTONS)) {
+      messageBlocks = createSlackButtonMessage((ChoicesResponseMessage) message);
+    } else {
+      messageBlocks = createSlackStaticSelectMessage((ChoicesResponseMessage) message);
+    }
+    logger.info(messageBlocks.toString());
```

Comment:
should be `debug` level log

Reply:
Overall if you are printing the messages, they should be debug level else it will fill up the logs and it will be too noisy.